src/services/facilityService.js

```diff
@@ -0,0 +1,88 @@
+import db from '../models';
+import { Helpers } from '../utils';
+
+const {
+  Facility, sequelize, RoomCategory, Room, FacilityAmenity, Amenity
+} = db;
+const { extractArrayRecords, updateCollection } = Helpers;
+
+/**
+ * A collection of methods that handles the database interactions
+ * for managing a Facility as an entity of the App.
+ *
+ * @class FacilityService
+ */
+class FacilityService {
+  /**
+   * Checks the room category value and creates a new record for any category added
+   * by user.
+   * @static
+   * @param {array} rooms - An array of rooms of a specific facility.
+   * @returns {Promise<Array>} A promise object with updated room properties.
+   * @memberof FacilityService
+   */
+  static async sortRoomCategory(rooms) {
+    const updatedRooms = rooms.map(async (room) => {
+      const { label } = await RoomCategory.findByPk(room.roomCategoryId);
```

Comment:
We should stick with destructuring
```suggestion
      const { label } = await findByPk(room.roomCategoryId);
```

Reply:
Thanks, @Segun-Ogundipe. Notice how `db` was destructed to have the models necessary for implementing the service methods herein. A second layer of destructing in this context doesn't feel semantically correct as it makes it more difficult to know which model findByPk belongs, reducing the readability of the code.

Reply:
What about 
```const { ..., RoomCategory: { findByPK }, ... } = dB;
``` 
?

Reply:
Thanks, @Baystef . As I said in my earlier response, the issue is not how to destruct but how it affects the readability of the code. But if the team would rather we destruct in this context I would be happy to amend the current PR. Kindly advise.

Reply:
Yeah, I totally understand. We've had such patterns of destructuring in the past PRs, so I think it should be familiar with fellow teammates already hopefully. Whether you change it or not is not left for only me to decide though so you can as well just leave it as it is. Great job @daylay92

Reply:
Thanks @Baystef

Reply:
> What about
> `const { ..., RoomCategory: { findByPK }, ... } = dB;`?

I really like this approach to de-structuring our imports.